packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -326,6 +327,7 @@ const BlockInspectorSingleBlock = ( { clientId, blockName } ) => {
 				className={ blockInformation.isSynced && 'is-synced' }
 			/>
 			<BlockVariationTransforms blockClientId={ clientId } />
+			{ blockName === 'core/query' && <CreateNewPostLink /> }
```

Comment:
This is very specific to `core/query` block and is probably better not to live in `block-editor` package.

An alternative, that is in need in general, is a new SlotFill inside `BlockCard` component. We could also consider just relocating the specific link inside `Query Loop` settings or something.

Reply:
Thanks for questioning this @ntsekouras 👍 

Initially, when the block inspector tabs were introduced we [explored the addition of a SlotFill to the BlockCard](https://github.com/WordPress/gutenberg/pull/45437). That approach was discarded due to concerns that 3rd parties could and would abuse the slot to promote their controls to higher prominence than warranted. We can't really lock down the SlotFill to internal use only as anyone that determines the name will be able to render fills to it.

There is also a precedent with the BlockCard component having `core/navigation` block-specific code which led to the current approach proposed in this PR.

As for relocating the link elsewhere, I'm not sure there is much appetite for that. In the [original issue](https://github.com/WordPress/gutenberg/issues/23461) that added this link, there were struggles in finding a proper home for the link and this was the only place that was agreed upon.

Would you have any objections to proceeding to land this fix and have the potential relocation of the link continue in a follow-up?

Reply:
Quite a lot have changed in UI since the introduction of this link, so maybe before proceeding with this approach we could have some extra input from @jameskoster or @jasmussen maybe, if there is a place we could relocate the link and not avoid adding the specific block functionality in `block-editor` package.

Reply:
I'll refer to Jay if he has input, he's deeper into the query stuff. But from a simple before and after, the after looks like an improvement. That's mostly ignoring the question of whether we need this link in the first place, or whether there's a better interface for adding new posts.

Reply:
If we want to keep the link then I'd agree with Joen, this is a visual improvement. But the UX leaves a lot to be desired:

1. It ejects you from the current edit state which is particularly jarring when you get sent from the site editor to the post editor.
2. "For this feed" is often untrue.
3. It's a bit tricky to interpret when editing templates like Archive, Category, Tag.

The whole thing ideally needs a fresh design exploration. But as low-hanging-fruit follow-ups I'd suggest updating the label to read "Add new post" which would address point 2 above. Bonus points if we can make it post type aware. E.g. if I configured a custom query to display `product`s, then the label would read "Add new product".

Reply:
I've tweaked the wording of the new post link as suggested to "Add new post". 

Making it display the translated name for the post type might be more involved as I don't think we can simply pull out the post type data from the `@wordpress/core-data` store within the block editor package. I could look into that as a follow-up to this though.

Reply:
Excellent thoughts Jay. That reminds me that we have an unfinished task for the split inspector followup, and those notes may be relevant in revisiting this:

* Explore a semantic slot, an ellipsis menu next to e.g. the query loop title.
* Ellipisis menu can have a third party hook for things like a premium block “badge”,  “this is a new block” unread dot, links to meta actions (add new post, etc).
* Explore a “Styles” panel revamp that includes more context for the Styles tab, help text, variations, copy styles, etc.

Reply:
> Making it display the translated name for the post type might be more involved as I don't think we can simply pull out the post type data from the @wordpress/core-data store within the block editor package

I think you can do `select( 'core' ).getPostType( postType )` to get the post type object, which includes the labels:
```
{
    "name": "Posts",
    "singular_name": "Post",
    "add_new": "Add New",
    "add_new_item": "Add New Post",
    "edit_item": "Edit Post",
    "new_item": "New Post",
    "view_item": "View Post",
    "view_items": "View Posts",
    "search_items": "Search Posts",
    "not_found": "No posts found.",
    "not_found_in_trash": "No posts found in Trash.",
    "parent_item_colon": null,
    "all_items": "All Posts",
    "archives": "Post Archives",
    "attributes": "Post Attributes",
    "insert_into_item": "Insert into post",
    "uploaded_to_this_item": "Uploaded to this post",
    "featured_image": "Featured image",
    "set_featured_image": "Set featured image",
    "remove_featured_image": "Remove featured image",
    "use_featured_image": "Use as featured image",
    "filter_items_list": "Filter posts list",
    "filter_by_date": "Filter by date",
    "items_list_navigation": "Posts list navigation",
    "items_list": "Posts list",
    "item_published": "Post published.",
    "item_published_privately": "Post published privately.",
    "item_reverted_to_draft": "Post reverted to draft.",
    "item_scheduled": "Post scheduled.",
    "item_updated": "Post updated.",
    "item_link": "Post Link",
    "item_link_description": "A link to a post.",
    "menu_name": "Posts",
    "name_admin_bar": "Post"
}
```

Reply:
Is that not the `@wordpress/core-data` store? I did search the codebase for `getPostType` and only saw uses of the core-data store which is restricted in the block-editor package.

Reply:
Ah yes, good point. 🤔